### PR TITLE
feature(champion path): add support  of "Шлях чемпіонів" (ChampionPath)  for Workshop

### LINKED ISSUE #1865: Feature: Implement modifications to Workshop for Ministry of Sport
## Task Description
1.  A new optional checkbox field, "Шлях чемпіонів" (Path of Champions), is added to the workshop creation/editing form.
2.  This checkbox is **only visible** on the user interface when the workshop's subordination is set to "Мінспорт" (Ministry of Sport).
3.  When "Мінспорт" subordination is selected:
    *   The "Тип гуртка" (`WorkshopType`) field becomes read-only (disabled).
    *   The value of "Тип гуртка" is automatically set to "Секція" (`Section`).
4.  If the subordination is changed from "Мінспорт" to any other value, the "Тип гуртка" field becomes editable again, and the "Шлях чемпіонів" checkbox is hidden.
5.  If a workshop is not under "Мінспорт" subordination, the "Шлях чемпіонів" property must be `false`.
6.  The value of the "Шлях чемпіонів" checkbox is saved to the database and indexed in Elasticsearch for filtering and search purposes.

### Backend Tasks

1.  Database Schema
    *   Add a new non-nullable boolean column to the `Workshops` table (e.g., `IsChampionPath`) with a `default` value of `false`.
    *   Generate and apply the corresponding database migration.

2.  Domain and Data Models
    *   Update the `Workshop` entity to include the new `IsChampionPath` property.
    *   Add the `IsChampionPath` property to the relevant DTOs used for creating, updating, and retrieving workshop data (e.g., `WorkshopBaseDto`, `WorkshopDto`).
    *   Update the Elasticsearch model `WorkshopES` to include the `IsChampionPath` property and ensure it's populated during data synchronization.

3.  Business Logic and Validation
    *   In the workshop creation and update services, implement the following validation logic:
        *   If the workshop's `InstitutionHierarchyId` corresponds to "Мінспорт" subordination:
            *   Enforce that the `WorkshopType` must be `Section`.
        *   If the workshop's `InstitutionHierarchyId` is **not** "Мінспорт":
            *   Ensure the `IsChampionPath` property is always set to `false`, ignoring any value that might be sent from the client.

### Frontend Tasks

1.  Data Models
    *   Update the `Workshop` TypeScript interface(s) to include the new optional boolean property (e.g., `isChampionPath?: boolean`).

2.  Workshop Form Component
    *   Conditional Field Visibility
        *   Add a new checkbox with the label "Шлях чемпіонів".
        *   Use component logic to show this checkbox *only if* the selected subordination is "Мінспорт".
    *   Subordination Change Logic:
        *   Subscribe to changes in the subordination selection dropdown.
        *   When the subordination is set to "Мінспорт":
            *   Patch the form to set the `WorkshopType` value to "Секція" (`Section`).
            *   Disable the `WorkshopType` form control to make it read-only.
        *   When the subordination changes from "Мінспорт" to another value:
            *   Enable the `WorkshopType` form control.
            *   Hide the "Шлях чемпіонів" checkbox and ensure its form value is set to `false`.

3.  API Integration
    *   Ensure the new `isChampionPath` property is included in the payload sent to the backend when creating or updating a workshop.

### PR DESCRIPTION
**Pull request by synapsetron**
### 🧩 Summary

This pull request introduces the **"Шлях чемпіонів" (ChampionPath)** functionality to the Workshop domain. It adds an optional checkbox that is visible and applicable only when a Workshop's subordination is set to **"Мінспорт"**. The implementation includes updates across the backend stack—models, validation, database schema, tests, and Elasticsearch indexing.

---

#### ✅ Backend Features
- **Added `IsChampionPath` boolean** field to `Workshop` domain model, DTOs, and Elasticsearch model.
- **Created and applied database migration**: adds a non-nullable `IsChampionPath` column with a default value of `false`.
- **Validation logic implemented**:
  - If `InstitutionHierarchyId` indicates "Мінспорт":
    - `WorkshopType` is enforced to be `"Секція"` and made read-only on UI level.
    - `IsChampionPath` is allowed.
  - If **not** "Мінспорт":
    - Any value sent for `IsChampionPath` is ignored, and stored as `false`.

#### 🔍 Elasticsearch
- `WorkshopES` model extended to support `IsChampionPath`.
- Ensured syncing logic populates the new property correctly.

#### 🧪 Tests
- Added unit tests for:
  - **Validation logic** across `CreateV1`, `CreateV2`, `UpdateV1`, and `UpdateV2`.
  - **WorkshopDraftService** for `createDraft`, `updateDraft`.
  - **Edge cases** involving subordination changes and their effect on `IsChampionPath`.

---
### 🧪 How to Test

1. Create or edit a workshop with `subordination = "Мінспорт"`.
2. Observe that:
   - `WorkshopType` is pre-set to `"Секція"` and disabled.
   - `"Шлях чемпіонів"` checkbox is visible and can be checked.
3. Change subordination to another value:
   - `"Шлях чемпіонів"` disappears.
   - `WorkshopType` becomes editable.
4. Validate changes in the database and Elasticsearch index.

### 📌 Notes

- The UI adjustments for field visibility and interactivity (e.g., disabling `WorkshopType`, showing checkbox) are assumed to be handled on the frontend separately.
- This PR strictly ensures **backend enforcement** for the correctness and integrity of `IsChampionPath`.

---

## Summary by CodeRabbit

* **New Features**
  * Introduced a new "IsChampionPath" flag for workshops, automatically set for workshops linked to the "Мінспорт" institution.
  * The "IsChampionPath" property is now consistently handled and visible across workshop creation, update, and draft flows.
  * Added configuration support for defining the Ministry of Sport institution title.

* **Bug Fixes**
  * Enhanced validation to ensure correct workshop type and champion path status assignment based on institution hierarchy.

* **Tests**
  * Expanded test coverage to verify behavior of the "IsChampionPath" flag and institution-based workshop type adjustments.

### CODE REVIEW

File: OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs [hunk 1]
```diff
@@ -1322,6 +1344,36 @@ private async Task ValidateLanguageExists(long LanguageOfEducationId)
...
+        if (institutionHierarchyId == null)
+        {
+            throw new InvalidOperationException("InstitutionHierarchyId cannot be null.");
+        }
+
+        // Get information about the institution hierarchy
+        var institutionHierarchyDto = await institutionHierarchyService.GetById(institutionHierarchyId.Value).ConfigureAwait(false);
+
+        // If the institution is "Мінспорт", set workshopType and isChampionPath; otherwise, use default values
+        var isChampionPath = institutionHierarchyDto.Institution.Title.Equals("Мінспорт", StringComparison.OrdinalIgnoreCase);
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs]: the title might change, so hardcoding is not an option. It is better to extract this to app settings, like:

```jsonc
{
  // ... other options
  "InstitutionSettings": {
    "MinistryOfSportTitle": "Мінспорт"
  }
}
```

or rather

```jsonc
{
  // ... other options
  "InstitutionSettings": {
    "MinistryOfSportInstitutionId": "<current guid here>"
  }
}
```

as the id might never change. And use the options pattern how we do with other settings

File: OutOfSchool/OutOfSchool.BusinessLogic/Services/WorkshopDrafts/WorkshopDraftService.cs [hunk 2]
```diff
@@ -1172,6 +1173,34 @@ private async Task SetLanguageNameOrThrow(WorkshopV2Dto dto)
...
+            throw new InvalidOperationException($"InstitutionHierarchy with ID = {dto.InstitutionHierarchyId} was not found.");
+        }
+
+        dto.IsChampionPath = institutionHierarchy.Institution.Title.Equals("Мінспорт", StringComparison.OrdinalIgnoreCase);
+
+        if (dto.IsChampionPath)
+        {
+            dto.WorkshopType = WorkshopType.Section;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (DmyMi) [reply OutOfSchool/OutOfSchool.BusinessLogic/Services/WorkshopDrafts/WorkshopDraftService.cs]: @synapsetron technically there should be no such case, but some mistakes can be made via manual database edit. So probably not so bad to do as rabbit suggests

**Review by DmyMi — APPROVED** [review]
